docs/DafnyRef/Attributes.md

```diff
@@ -147,8 +147,7 @@ It means that the post-condition may be assumed to be true
 without proof. In that case also the body of the function or
 method may be omitted.
 
-The `{:axiom}` attribute is internally equivalent to `{:verify false}` and is also used for the internal `reveal_*`
-lemmas as shown in Section [#sec-opaque].
+The `{:axiom}` attribute is internally equivalent to `{:verify false}`.
```

Comment:
By the way, why is the compiler complaining about body-less functions?  Given that functions are ghost, I wouldn't expect the compiler to look at them at all.

Reply:
I think this is to enable quick prototyping while not being bothered by "missing body" errors. This check has to be done somewhere.
But if we run the compiler's diagnostics as part of the resolving diagnostics, we might get a "missing body" error which would prevent us from triggering verification. I'm just thinking.

Reply:
Declaring a function or lemma without giving it a body is like adding a body with an `assume false` -- it lets users of the function or lemma assume the specification, but there's nothing to ensure that the specification holds. If a user accidentally left off the body of the function or lemma, then it could be that the program is not correct even if everything else in the program verifies. So, _something_ in the Dafny pipeline should give an error about this.

One choice would be to let the verifier complain about it. A different choice would be to make the compiler complain about it. A third choice would be to have another pass in the Dafny pipeline, perhaps between the verifier and compiler, that complains about it. It's nice _not_ to give the error in the verifier, because then users don't have to get verification errors until they start writing the body of the function or lemma. And it _is_ convenient to let the compiler give the error, because the compiler already has to check that all functions and methods that it's supposed to turn into code have bodies. So, the choice made in the Dafny pipeline is for the compiler for look for and complain about body-less entities, including body-less _ghost_ entities.